docs/adr/0007-modelling-history.md

```diff
@@ -0,0 +1,272 @@
+# 7. Modelling history
+
+Date: 2018-12-17
+
+## Context
+
+Using terminology as defined in [ADR-3](../0003-initial-domain-modelling.md):
+
+- **Document**: A piece of content in a particular locale that can have many
+versions;
+- **Revision**: A particular edit of a piece of content, represents a document at
+a specific point in time;
+- **Edition**: A particular revision that was published to GOV.UK or is the
+current draft of the document.
+
+Content Publisher currently has a concept of a document and not editions or
+revisions. A consequence of this is that an edit to a document overwrites the
+previous state of the document. Since a document can exist in two forms, draft
+and live, and Content Publisher only knows of one we hit problems when
+knowledge of both would be useful.
+
+The suggestion is that we enhance Content Publisher to support the concepts of
+Document, Edition and Revision. The features that we'd like this to enable
+are:
+
+For a document with a live version and a draft:
+
+- The ability to discard the draft and revert the current version in Content
+  Publisher back to the live version (including all images and attachments) to
+  be in sync with Publishing API;
+- The ability to have the details of the live version, most noticeably state
+  and URL so this can be presented to users;
+- Ability to re-present either the live or draft version to Publishing API
```

Comment:
Is there an alternative to re-present? Perhaps 'resubmit'

Reply:
Sure, it's not meant to be for a user but for GOV.UK developers.

Some of the scenarios where this has been needed in the past:

- Bug found in means of publishing to Publishing API so decided documents are invalid
- The rather miserable decision that a new field is needed on a document type and want all old documents to have it

Some realistic scenarios for us would probably be if we found out a contact was updated or if there was an issue with one of the new govspeak things we use.

On the more semantic side of things since Content Publisher is the canonical source of data it should always have sufficient information to re-populate Publishing API.